plugin/diagnostics.py

```diff
@@ -3,51 +3,56 @@
 
 from .core.protocol import Diagnostic, Point, Range
 from .core.registry import windows
-from .core.typing import List, Dict, Tuple
+from .core.sessions import SessionBufferProtocol
+from .core.typing import List, Dict, Tuple, Generator
 
 
-def view_diagnostics(view: sublime.View) -> Dict[str, List[Diagnostic]]:
+def view_diagnostics(
+    view: sublime.View
+) -> Generator[Tuple[SessionBufferProtocol, List[Diagnostic]], None, None]:
+    """XXX: Remove this function"""
     window = view.window()
     if window:
         manager = windows.lookup(window)
         listener = manager.listener_for_view(view)
         if listener:
-            return listener.diagnostics_async()
-    return {}
+            yield from listener.diagnostics_async()
 
 
 def filter_by_point(
-    file_diagnostics: Dict[str, List[Diagnostic]],
+    file_diagnostics: Generator[Tuple[SessionBufferProtocol, List[Diagnostic]], None, None],
     point: Point
 ) -> Tuple[Dict[str, List[Diagnostic]], Range]:
+    """XXX: Move this odd function to DocumentSyncListener"""
     diagnostics_by_config = {}
     extended_range = Range(point, point)
-    for config_name, diagnostics in file_diagnostics.items():
+    for sb, diagnostics in file_diagnostics:
```

Comment:
By first glance it is not clear what `sb` means. Maybe rename it to session_buffer?